<launch>

    <arg name="marker_size" default="5.6" />
    <arg name="max_new_marker_error" default="0.08" />
    <arg name="max_track_error" default="0.2" />
    <arg name="cam_image_topic" default="/usb_cam/image_raw" />
    <arg name="cam_info_topic" default="/usb_cam/camera_info" />
    <arg name="output_frame" default="/usb_cam" />
    <arg name="mname" default="zumy7a" />

    <node name="usb_cam" pkg="usb_cam" type="usb_cam_node"  output="screen" >
      <param name="video_device" value="/dev/video0" />
      <param name="image_width" value="1280" />
      <param name="image_height" value="720" />
      <param name="pixel_format" value="mjpeg" />
      <param name="camera_frame_id" value="usb_cam" />
      <param name="io_method" value="mmap" />
      <param name="camera_info_url" value="file:///home/cc/ee106b/sp16/class/ee106b-am/ros_workspaces/project1/yml/lifecam.yml" />
    </node>

    <node name="ar_track_alvar" pkg="ar_track_alvar" type="individualMarkersNoKinect" respawn="false" output="screen" args="$(arg marker_size) $(arg max_new_marker_error) $(arg max_track_error) $(arg cam_image_topic) $(arg cam_info_topic) $(arg output_frame)" />

    <include file="$(find odroid_machine)/launch/remote_zumy.launch">
      <arg name="mname" value="zumy7a" />
    </include>

 <!--  <node machine="$(arg mname)" ns="zumy7a" name="zumy_ros_bridge" pkg="ros_zumy" type="zumy_ros_bridge.py"  output="screen" >
  </node> -->

    <node name="controller" pkg="pid" type="controller" ns="l_pid" output="screen" >
      <param name="node_name" value="l_pid" />
      <!-- K values for the left wheel -->
      <param name="Kp" value="0.0" /> 
      <param name="Ki" value="0.0" />
      <param name="Kd" value="0.0" />
      <param name="upper_limit" value="10" />
      <param name="lower_limit" value="-10" />
      <param name="windup_limit" value="10" />
      <param name="diagnostic_period" value="0.25" />
      <param name="max_loop_frequency" value="100.0" />
      <param name="min_loop_frequency" value="100.0" />
      <remap from="setpoint" to="/l_pid/setpoint" />
     </node>

    <node name="servo_sim_node" pkg="pid" type="plant_sim" ns="l_pid" output="screen" >
      <param name="plant_order" value="2" />
    </node>

    <node name="controller" pkg="pid" type="controller" ns="r_pid" output="screen" >
      <param name="node_name" value="r_pid" />
      <!-- K values for the right wheel -->
      <param name="Kp" value="0.0" />
      <param name="Ki" value="0.0" />
      <param name="Kd" value="0.0" />
      <param name="upper_limit" value="10" />
      <param name="lower_limit" value="-10" />
      <param name="windup_limit" value="10" />
      <param name="diagnostic_period" value="0.25" />
      <param name="max_loop_frequency" value="100.0" />
      <param name="min_loop_frequency" value="100.0" />
      <remap from="setpoint" to="/r_pid/setpoint" />
     </node>

    <node name="servo_sim_node" pkg="pid" type="plant_sim" ns="r_pid" output="screen" >
      <param name="plant_order" value="2" />
      <param name="reverse_acting" value="true" />
    </node>


</launch>